server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -405,8 +405,8 @@ public Query rangeQuery(
                         query,
                         new ApproximatePointRangeQuery(
                             field,
-                            NumberType.HALF_FLOAT.encodePoint(l),
-                            NumberType.HALF_FLOAT.encodePoint(u),
+                            HalfFloatPoint.pack(new float[] { l }).bytes,
```

Comment:
nit: I am wondering if `HalfFloatPoint.pack(l).bytes` looks cleaner instead? Is there any advantage of explicitly initializing the array?